exporter/trace/jaeger/jaeger_test.go

```diff
@@ -15,24 +15,90 @@
 package jaeger
 
 import (
+	"context"
 	"encoding/binary"
 	"sort"
 	"testing"
 	"time"
 
 	"go.opentelemetry.io/api/key"
 
-	apitrace "go.opentelemetry.io/api/trace"
-
 	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/assert"
 	"google.golang.org/grpc/codes"
 
+	apitrace "go.opentelemetry.io/api/trace"
+	sdktrace "go.opentelemetry.io/sdk/trace"
+
 	"go.opentelemetry.io/api/core"
 	gen "go.opentelemetry.io/exporter/trace/jaeger/internal/gen-go/jaeger"
 	"go.opentelemetry.io/sdk/export"
 )
 
-// TODO(rghetia): Test export.
+func TestNewExporter(t *testing.T) {
+	const (
+		collectorEndpoint = "http://localhost"
+		serviceName       = "test-service"
+		tagKey            = "key"
+		tagVal            = "val"
+	)
+	// Create Jaeger Exporter
+	exp, err := NewExporter(
+		WithCollectorEndpoint(collectorEndpoint),
+		WithProcess(Process{
+			ServiceName: serviceName,
+			Tags: []core.KeyValue{
+				key.String(tagKey, tagVal),
+			},
+		}),
+	)
+
+	assert.NoError(t, err)
+	assert.EqualValues(t, serviceName, exp.process.ServiceName)
+	assert.Len(t, exp.process.Tags, 1)
+}
+
+func TestNewExporterShouldFailIfCollectorEndpoingEmpty(t *testing.T) {
+	_, err := NewExporter(
+		WithCollectorEndpoint(""),
+	)
+
+	assert.Error(t, err)
+}
+
+func TestExporter_ExportSpan(t *testing.T) {
+	const (
+		collectorEndpoint = "http://localhost"
+		serviceName       = "test-service"
+		tagKey            = "key"
+		tagVal            = "val"
+	)
+	// Create Jaeger Exporter
+	exp, err := NewExporter(
+		WithCollectorEndpoint(collectorEndpoint),
+		WithProcess(Process{
+			ServiceName: serviceName,
+			Tags: []core.KeyValue{
+				key.String(tagKey, tagVal),
+			},
+		}),
+	)
+
+	assert.NoError(t, err)
+
+	exp.RegisterSimpleSpanProcessor()
+	tp, err := sdktrace.NewProvider(
+		sdktrace.WithConfig(sdktrace.Config{DefaultSampler: sdktrace.AlwaysSample()}),
+		sdktrace.WithSyncer(exp))
+
+	assert.NoError(t, err)
+
+	apitrace.SetGlobalProvider(tp)
+	_, span := apitrace.GlobalProvider().GetTracer("test-tracer").Start(context.Background(), "test-span")
+	span.End()
+
+	assert.True(t, span.SpanContext().IsValid())
```

Comment:
It would be nice to have a mock Jaeger collector and verify that it receives the span.

Reply:
@rghetia fixed, if it makes sense, i will also add similar tests to the agent as well

Reply:
looks good.

Reply:
done and added tests for the agent as well